### ISSUE TITLE
deCONZ addon stops itself randomly

### ISSUE THREAD

**Issue by SiliconAvatar** [home-assistant/addons#1546]
## The problem

The deCONZ addon seems to randomly stop itself without warning. It can be started again normally and work just fine, but then will stop again some indeterminate time later. It's happened to me twice so far, 7 days apart. However there are many other reports of this behaviour on the [community forum](https://community.home-assistant.io/t/deconz-official-thread/100504).

## Environment

- Add-on with the issue: deCONZ 
- Add-on release with the issue: Version 6.2.3 
- Last working add-on release (if known): 6.2.2 (?)
- Operating environment (OS/Supervised): Supervised

arch | x86_64
-- | --
chassis | vm
dev | false
docker | true
docker_version | 19.03.8
hassio | true
host_os | Ubuntu 18.04.4 LTS
installation_type | Home Assistant Supervised
os_name | Linux
os_version | 5.4.0-42-generic
python_version | 3.8.3
supervisor | 235
timezone | America/New_York
version | 0.114.1
virtualenv | false

## Problem-relevant configuration

N/A

## Traceback/Error logs

[Photo of log output.](https://i.imgur.com/EzCXIQE.png), I unfortunately did not copy out the text before starting the addon again.

**Comment by xbmcnut** [comment]
I too have the same issue on Intel NUC with 0.114.3 with deCONZ 6.2.3 and firmware 0x26580700 on a Conbee II. My only anomaly is that I've recently added a 2nd Conbee and ZHA so I've removed both this evening to see if things settle down. N.B I also run a Aeon Z-stick on another USB port is that could be a factor.

**Comment by mkono87** [comment]
> Look like a bug in deCONZ. It crash some times. I think the solution will be a watchdog wich restart the add-on if they crash

Extra container installed outside through something like portainer?

**Comment by pvizeli** [comment]
https://github.com/home-assistant/supervisor/pull/1970

**Comment by vassilis-panos** [comment]
Unfortunately restarting the addon in case it crashes will create more problems for me. Assuming we have activated the security alarm and restarted the addon then the state of the sensors will change and activate it.
Does anyone know what was the latest stable version of the addon to restore it? Thanks.

**Comment by pvizeli** [comment]
I guess the .75 was very stable - otherwise, you can also look into the internal ZHA integration of home assistant.

**Comment by austwhite** [comment]
> Unfortunately restarting the addon in case it crashes will create more problems for me. Assuming we have activated the security alarm and restarted the addon then the state of the sensors will change and activate it.
> Does anyone know what was the latest stable version of the addon to restore it? Thanks.

My DeConz was stable under 0.112.x.  Maybe try that one.
I only experienced crashes in 0.114.x.   I don't recall 0.113.x as I wasn't on that version for long

**Comment by SiliconAvatar** [comment]
Just an update that this issue is ongoing, it just crashed again today.

```
18:31:36:214 Current channel 25
18:31:36:227 Device TTL 1554 s flags: 0x7
[18:31:36] WARNING: Your direct VNC access is not protected!
[18:31:36] INFO: Starting VNC server (local/no)...
2020/08/31 22:31:36 [notice] 320#320: signal 15 (SIGTERM) received from 314, exiting
2020/08/31 22:31:36 [notice] 1734#1734: exiting
2020/08/31 22:31:36 [notice] 1734#1734: exit
2020/08/31 22:31:36 [notice] 320#320: signal 17 (SIGCHLD) received from 1734
2020/08/31 22:31:36 [notice] 320#320: worker process 1734 exited with code 0
2020/08/31 22:31:36 [notice] 320#320: exit
In exit
Closing socket listening at 127.0.0.1:5901
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
```

**Comment by qtjupsrkb** [comment]
Weirdly, it hadn't happened to me for a few days until today.

```
05:13:21:999 don't close database yet, keep open for 900 seconds
[05:13:25] WARNING: Your direct VNC access is not protected!
[05:13:25] INFO: Starting VNC server (local/no)...
2020/08/31 19:13:25 [notice] 325#325: signal 15 (SIGTERM) received from 317, exiting
2020/08/31 19:13:25 [notice] 1676#1676: exiting
2020/08/31 19:13:25 [notice] 1676#1676: exit
In exit
Closing socket listening at 127.0.0.1:5901
2020/08/31 19:13:25 [notice] 325#325: signal 17 (SIGCHLD) received from 1676
2020/08/31 19:13:25 [notice] 325#325: worker process 1676 exited with code 0
2020/08/31 19:13:25 [notice] 325#325: exit
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting
```

**Comment by austwhite** [comment]
Did the most recent crash happen just after updating Home Assistant?
Mine had been stable for a week but then crashed three times after updating

**Comment by SiliconAvatar** [comment]
> Did the most recent crash happen just after updating Home Assistant?

Not really, I last updated when 0.114.4 was released on the 26th, and deCONZ had been running since the 21st until the crash tonight.

**Comment by pergolafabio** [comment]
Same here , happened again today after uptime of 6 days, 114.4.  Frustrating

**Comment by jdbrookes** [comment]
Deconz suddenly stopped on my tinkerboard after running fine for 10 days: happened at 23:45 last night (GMT+1). 114.3. Never happened before. Deconz firmware up to date on conbee II. 

Is it a coincidence that a lot of users saw this mysterious issue at around the same time last night?

**Comment by pergolafabio** [comment]
Same timestamp here..., Also about 10 days, and indeed around 23.45

**Comment by Qpernicus** [comment]
I have the same issue and my log looks like most here, nothing that indicates a reason for shutting down.

**Comment by aerodinamica** [comment]
Same issue here. Running HA in a Proxmox VM (Intel x86 host). This has happened twice now. I did abruptly disconnect three devices yesterday, possibly disrupting the Zigbee network. That could just be a coincidence though.

**Comment by falcon2k16** [comment]
Same issue happened here. This started happening in August for me but the first two times I never really thought that it could be an issue as I didn't check the logs but just started the add-on to keep the house going. 

I'm running HA Supervised on Ubuntu 20.04 LTS with a DeConz and Z-Wave stick. 

Happened 23:43 as you could see from logs screenshot below 
![deconz_issue](https://user-images.githubusercontent.com/17091949/91814761-d7ba8080-ec34-11ea-83a3-f7cd05ca7310.jpg)
![deconz_issue_2](https://user-images.githubusercontent.com/17091949/91814772-d8531700-ec34-11ea-9ca8-9ad5eca5fd14.jpg)

**Comment by cllatser** [comment]
After 10 days it has stopped again. The last thing the addon did was something related to the **VCN server**. Leave the VCN connection as it was before, separate. Whoever wants to enter through VCN to install the client program. Thank you so much.



00:33:39:874 don't close database yet, keep open for 900 seconds
00:33:43:917 Bind response success for 0x842e14fffe51c5d5 ep: 0x01 cluster: 0x0B04
00:33:46:893 don't close database yet, keep open for 900 seconds
00:33:46:920 ZCL attribute report 0x680AE2FFFE7240CF for cluster: 0x0008, ep: 0x01, frame control: 0x08, mfcode: 0x0000 
00:33:47:947 no button map for: SML001 ep: 0x02 cl: 0x0402 cmd: 0x0A pl[0]: 000
00:33:47:947 ZCL attribute report 0x001788010862E71F for cluster: 0x0402, ep: 0x02, frame control: 0x08, mfcode: 0x0000 
00:33:51:965 don't close database yet, keep open for 900 seconds
00:33:57:914 don't close database yet, keep open for 900 seconds
00:34:02:971 don't close database yet, keep open for 900 seconds
00:34:04:461 Set sensor check interval to 1000 milliseconds
00:34:07:876 Current channel 15
00:34:07:884 Device TTL 6180 s flags: 0x7
[00:34:08] WARNING: Your direct VNC access is not protected!
**[00:34:08] INFO: Starting VNC server (local/no)...**
In exit
Closing socket listening at 127.0.0.1:5901
2020/08/31 22:34:08 [notice] 325#325: signal 15 (SIGTERM) received from 316, exiting
2020/08/31 22:34:08 [notice] 1560#1560: exiting
2020/08/31 22:34:08 [notice] 1560#1560: exit
2020/08/31 22:34:08 [notice] 325#325: signal 17 (SIGCHLD) received from 1560
2020/08/31 22:34:08 [notice] 325#325: worker process 1560 exited with code 0
2020/08/31 22:34:08 [notice] 325#325: exit
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.

**Comment by pergolafabio** [comment]
My guess it's indeed the new vnc server, that was added in last release, gonna disable it

**Comment by gribber** [comment]
It did not die to me last night as it has done before, but I did _not_ have a open VNC connection to it this time as I usually do

**Comment by pergolafabio** [comment]
But last add-on version included a web bases vnc server, maybe that one is the culprit

**Comment by AlexSanti** [comment]
The same problem

2020/08/31 22:38:39 [notice] 320#320: signal 15 (SIGTERM) received from 317, exiting
In exit
Closing socket listening at 127.0.0.1:5901
2020/08/31 22:38:39 [notice] 1825#1825: exiting
2020/08/31 22:38:39 [notice] 1825#1825: exit
2020/08/31 22:38:39 [notice] 320#320: signal 17 (SIGCHLD) received from 1825
2020/08/31 22:38:39 [notice] 320#320: worker process 1825 exited with code 0
2020/08/31 22:38:39 [notice] 320#320: exit
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.

**Comment by pergolafabio** [comment]
Hmm , seems its not possible to disable ingress VNC only, this was last added, maybe caused by that

**Comment by Dinth** [comment]
Funny enough, yes ter it crashed for me around 23.45 too

**Comment by pvizeli** [comment]
You can update to beta (channel beta) and use the new function `watchdog` on the add-on

**Comment by pergolafabio** [comment]
yeah, already have my own automation now, that checks if deconz is runnng, if not => restart
but it shouldt happen, there is clearly something wrong witrh the addon, it stopped for  a lot of users around 23.45

**Comment by vassilis-panos** [comment]
It did not stop for me this time.. The only change I made was to delete the VNC port

**Comment by pergolafabio** [comment]
yeah, i think its caused by VNC with ingress
if you disable VNC port, can you still use VNC with the tool? like it was before? or if you disable VNC, is it for both ingress and tool ?

**Comment by austwhite** [comment]
> You can update to beta (channel beta) and use the new function `watchdog` on the add-on

Does this actually fix the cause of the crashing or does it just introduce a bandaid?

**Comment by pergolafabio** [comment]
no, it doenst fix, its just a watchdog, that will restart when stopped, a temp workaround, not a solution yet

**Comment by pierinho** [comment]
> yeah, already have my own automation now, that checks if deconz is runnng, if not => restart
> but it shouldt happen, there is clearly something wrong witrh the addon, it stopped for a lot of users around 23.45

Can you share this automation? Tonight My Deconz stop working...

**Comment by pergolafabio** [comment]
offcourse
here is my sensor : 

```
  - platform: command_line
    name: deCONZ
    command: 'curl http://supervisor/addons/core_deconz/info -H "Authorization: Bearer $(printenv SUPERVISOR_TOKEN)" | jq ''{"state":.data.state,"newest_version":.data.version_latest,"current_version":.data.version,"available":.data.available}'''
    value_template: '{{ value_json.state }}'
    json_attributes:
    - available
    - newest_version
    - current_version
```

and here is my automation : 
 

```
- alias: Deconz Addon Stopped
  initial_state: 'on' 
  trigger:
    platform: state
    entity_id:
      - sensor.deconz
    from: started
    to: stopped
  action:
    - service: hassio.addon_start
      data:
        addon: "core_deconz" 
```

**Comment by migo-dev** [comment]
> @pergolafabio if you disable VNC port, can you still use VNC with the tool? like it was before? or if you disable VNC, is it for both ingress and tool ?

You can still access through the new web vnc. This does only disable the "old" vnc. I had the port removed ever since the one update where the addon couldn't start with a port in vnc. This night at 00:31 (CEST) my addon stopped for the first time - so also around the same time (~15 min) as yours and some of the other guys.

**Comment by pierinho** [comment]
> > @pergolafabio if you disable VNC port, can you still use VNC with the tool? like it was before? or if you disable VNC, is it for both ingress and tool ?
> 
> You can still access through the new web vnc. This does only disable the "old" vnc. I had the port removed ever since the one update where the addon couldn't start with a port in vnc. This night at 00:31 (CEST) my addon stopped for the first time - so also around the same time (~15 min) as yours and some of the other guys.

My addon stopped working at 00:33 Italian Time

**Comment by Dinth** [comment]
> offcourse
> here is my sensor :

Thanks for sharing your code.
Ive improved your automation a little bit:

```
automation:
  - alias: DeCONZ Addon Stopped
    initial_state: on
    trigger:
      platform: state
      entity_id:
      - sensor.deconz
      from: 'started'
      to: 'stopped'
    action:
    - service: notify.notifier_telegram
      data:
        title: 'DeCONZ addon'
        message: 'DeCONZ addon crashed - attempting restart...'
    - service: hassio.addon_start
      data:
        addon: "core_deconz" 
        action:
    - service: notify.notifier_telegram
      data_template:
        title: 'DeCONZ addon'
        message: >-
          {% if is_state('sensor.deconz', 'started') %}
          DeCONZ restarted successfully!
          {% else %}
          DeCONZ failed to restart!
          {% endif %}
```

**Comment by pergolafabio** [comment]
yeah, i also have some notifications, i dropped that part , also i have some popups :)
just posted the necesary :)

**Comment by Dinth** [comment]
> yeah, i also have some notifications, i dropped that part , also i have some popups :)
> just posted the necesary :)

The only thing it's missing is saving Deconz log upon crash, maybe someone is willing to improve it  further, as  ive got no idea how to do that? :)

**Comment by pergolafabio** [comment]
we  all know the log report, its always the same :) ,

**Comment by mkono87** [comment]
How exactly do you disable it? I have been using ingress so I never entered a port number. 
![chrome_n8eb7D1Eta](https://user-images.githubusercontent.com/35116314/91856335-70043580-ec34-11ea-95d2-340f0b72d3e0.png)

**Comment by pergolafabio** [comment]
Not sure, I have the same question actually, mine is also different, I only see the 5900 line, not the 8081

**Comment by arifroni** [comment]
> > offcourse
> > here is my sensor :
> 
> Thanks for sharing your code.
> Ive improved your automation a little bit:
> 
> ```
> automation:
>   - alias: DeCONZ Addon Stopped
>     initial_state: on
>     trigger:
>       platform: state
>       entity_id:
>       - sensor.deconz
>       from: 'started'
>       to: 'stopped'
>     action:
>     - service: notify.notifier_telegram
>       data:
>         title: 'DeCONZ addon'
>         message: 'DeCONZ addon crashed - attempting restart...'
>     - delay:
>         minutes: 2
>     - service: hassio.addon_start
>       data:
>         addon: "core_deconz" 
>         action:
>     - service: notify.notifier_telegram
>       data_template:
>         title: 'DeCONZ addon'
>         message: >-
>           {% if is_state('sensor.deconz', 'started') %}
>           DeCONZ restarted successfully!
>           {% else %}
>           DeCONZ failed to restart!
>           {% endif %}
> ```

BTW delay should be after the restarting the addon, not before. right?

**Comment by pergolafabio** [comment]
no, delay is not needed at all i think
he just wants a delay between stop and start  , you dont need it actually

**Comment by Dinth** [comment]
> no, delay is not needed at all i think
> he just wants a delay between stop and start , you dont need it actually

No, actually @arifroni is right here, i wanted to make a delay after restart and before the check if service is running.
Im going to edit my post

**Comment by pergolafabio** [comment]
ah ok 

good workaround for now, but still, needs to be look at , i hope a dev can look into it, or how can we see a more detailed log report of this addon ?

**Comment by dedepene** [comment]
Exact same thing has been happening rather haphazardly. The Deconz addon got murdered again just a day ago.

**Comment by brisketfingers** [comment]
Same thing happening to me. Nothing else in the logs indicates an issue.

[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting

Shutdown happens immediately after starting the container, consistently, I've had to revert deCONZ from a snapshot.

**Comment by megawubs** [comment]
Today I had the same issue. The line `03:06:09] WARNING: Your direct VNC access is not protected!` got me scared. I was under the assumption someone must have gotten access to the VNC and shut it down. 

> On community forum, in deconz thread, there is an example sensor, this will tell you when an add-on is stopped

Can you provide a link?

**Comment by pergolafabio** [comment]
That's not needed anymore, watchdog is now released in supervisor... Or scroll up

**Comment by dedepene** [comment]
crashed again, now getting lot's of these. Before, after the crash, and even after a successful addon restart

` 07:18:38 ERROR (MainThread) [pydeconz.websocket] Client connection error`

**Comment by tosion** [comment]
Just another piece of info for this issue.

I have two HA NUC image setups 200 km apart from each other. Another one has had this issue 4 times  (maybe in around a month or so including last night) another had this first time last night. Weird is that timing is within two minutes. I have watchdog enabled now.

![image](https://user-images.githubusercontent.com/48414735/93426113-105c8a00-f8c4-11ea-91d6-f13f4c873285.png)
![image](https://user-images.githubusercontent.com/48414735/93426237-50237180-f8c4-11ea-8469-053c373c91eb.png)

![image](https://user-images.githubusercontent.com/48414735/93426313-70ebc700-f8c4-11ea-9e15-9328ae7ad417.png)

**Comment by bmesuere** [comment]
I also had another crash. Seems like a strange coincidence that this always happens around the same moment for a lot of people.

**Comment by HammerNL89** [comment]
Didn't know this was an issue, I thought it was related to my setup. But for me it also crashed this night (03:16 NL): 
![down](https://user-images.githubusercontent.com/6350295/93434695-c75b0480-f8c8-11ea-956a-4420121e9e95.jpg)

**Comment by bmesuere** [comment]
Here in Belgium, my zigbee lights became unavailable at 3:10:52

**Comment by tosion** [comment]
And I am located in Finland so it happened at same time ...

**Comment by pergolafabio** [comment]
i normally have this issue also, except this time, but its because i did a complete reboot of my Hassos system, so the deconz addon was also restarted

so i think if its has an uptime for like several days, i think 7 , it crashes ...
maybe its because of the builtin vnc server ? maybe its a trial  :)

**Comment by jdbrookes** [comment]
Well I'm back 16 days later to report that deconz again mysteriously stopped last night, this time at 02:07 IST, so there's definitely an issue with some aspect of this addon that impacts a number of users at a common time.

Thanks to @pergolafabio for his excellent code snippets that monitored the addon, realised it had stopped, started it again, and sent me a notification to tip me off. I realise there is now a watchdog feature builtin to hassio, but I like this particular solution for keeping the show on the road. Otherwise there would have been no lights coming on automatically when the family got up this morning, which hurts the WAF.

**Comment by austwhite** [comment]
If it is critical for it to keep running for WAF remember to turn on the watchdog function.
It's not a fix, but at least it automatically restarts the service if it crashes.
That will keep the waf points up until a permanent fix is found

**Comment by pergolafabio** [comment]
is someone actually looking into this issue?

**Comment by austwhite** [comment]
@fenck as the one who generally looks after these add-ons so I expect he would be looking.  He is usually pretty good like that

**Comment by pvizeli** [comment]
We wait of the manufacture (deconz) to make an release which we can use. That is not an issue which we can solve by us.

**Comment by pergolafabio** [comment]
But do you know the root cause?

**Comment by cllatser** [comment]
Well how do I see that they did not solve it. I have switched to Deconz in Docker and this time it has not stopped me. The Addon is not maintained by Deconz, so you always depend on having someone from HA fix it.

**Comment by pergolafabio** [comment]
Running 6.3.0, my deconz container stopped twice an hour ago... 

Does anyone know already the cause? Is someone working on a fix?

**Comment by vukisz** [comment]
Mine crashed without restocking of plug in our even HA restart. Plug and unplug conbee helped.

**Comment by accelle17** [comment]
When this addon stops scenario happens, I usually noticed that my conbee 2 stick (fw 26580700 to 2665I0700) has reconnection logs on dmesg/syslog. I'm on NUC running proxmox 6.2 with hassos but I also saw this when I was on pi4.

**Comment by pergolafabio** [comment]
This night it stopped 3 times in a row, anyone else too? Is this issue actually under investigation?

**Comment by SiliconAvatar** [comment]
> This night it stopped 3 times in a row, anyone else too? Is this issue actually under investigation?

Crashed for me last night again at 10:48pm EST.
Only once that I'm aware of. I'm using an automation now to log that it's crashed and try to restart it, so I don't have the logs previous to the crash, but it's a safe bet it's the same as it was every other time, and the same as all the logs already posted here.

I'm not using the watchdog because that provides no indication (AFAIK) that it crashed, and I fear that people using it now are going to reduce the number of reports of this issue. The watchdog only hides the issue, not fix it.

**Comment by EmbeddedDevver** [comment]
Big warning to all users having this 'problem'!!!

I lost all my 'light' nodes for the Conbee 2 stick and it looks like it is not repairable and I need to go over and repair every single node in my house again.

I updated 1 week ago, before, it was running for more than 18 months stable with no issues. After the latests updates (both hassio and conbee plugin / firmware), I noticed that the plugin stopped randomly about 2 times a day. Everytime I pressed start again without investigating due lack of time. 

Today, I did the same but now when the plugin was loaded, I noticed that all of the 'light' switches were gone inside phoscon. 

I went back to the old backup of hassio, but apparently hassio doesn't backup the data on the conbee 2 stick itself. So my old backup isn't working anymore too with the conbee 2 usb stick. Also more recent versions of the last days are reverted, but still all the 'lights' are gone.

**Comment by pergolafabio** [comment]
That's not related at all to this issue, please create a new issue... This issue is just about the container that's stops randomly ;)

**Comment by EmbeddedDevver** [comment]
> That's not related at all to this issue, please create a new issue... This issue is just about the container that's stops randomly ;)

I think this is exactly the same issue. The container / plugin inside hassio stops every day randomly. I have to press start there again to run it. Isnt this the same where your issue is about

**Comment by pergolafabio** [comment]
I still think it's something else you describe, we are facing the stops not randomly, it's for most users always on same day/timestamp .... Not randomly every day

For me it's once a week, I think always on Sunday

**Comment by SiliconAvatar** [comment]
> We wait of the manufacture (deconz) to make an release which we can use. That is not an issue which we can solve by us.

So are you aware of what the issue is, which is causing this? Is it something being tracked on their issue repo?
Why are you sure it's the deCONZ software causing the issue, and not the way the application is being used/built inside the docker container, containing VNC components, etc. 
I'm not being augmentative, just wondering what that conclusion is based on.

From what I hear from other people running the deCONZ software on bare metal, they are not having this problem; which may indicate that it's not a problem with the application, but with this addon.

**Comment by pergolafabio** [comment]
> > We wait of the manufacture (deconz) to make an release which we can use. That is not an issue which we can solve by us.
> 
> So are you aware of what the issue is, which is causing this? Is it something being tracked on their issue repo?
> Why are you sure it's the deCONZ software causing the issue, and not the way the application is being used/built inside the docker container, containing VNC components, etc.
> I'm not being augmentative, just wondering what that conclusion is based on.
> 
> From what I hear from other people running the deCONZ software on bare metal, they are not having this problem; which may indicate that it's not a problem with the application, but with this addon.

wondering the same too, i dont see any issues posted at all on deconz github website about stopping of the deamon

**Comment by austwhite** [comment]
If this is an upstream issue/known issue please give github issue number and link.
The guys that maintain the DeConz/phoscon apps are very quick to fix bugs like this usually and there have now been 4 updates to deConz/Phoscon since this bug was first reported on home assistant.
This is ridiculous for a system which is pretty much the core functionality of an automation system.
Stop bouncing the blame and fix it.  If you can't, then roll back the add-on to the last working version.  That has been done before

**Comment by accelle17** [comment]
for those on proxmox, mind looking on your syslog if the conbee 2 is getting reconnected? This issue happened to me [before](https://community.home-assistant.io/t/deconz-official-thread/100504/1890) on pi 4. Moved to proxmox and NUC and it got escalated to hassos crashing after updating it to 4.13 and upgrading the addon to 6.3.0. It turned out that it was an existing proxmox issue, see links on [870](https://github.com/home-assistant/operating-system/issues/870). It has been 10 days since I reinstalled the addon and replaced my conbee 2 stick, no crash yet.

**Comment by pergolafabio** [comment]
the add-on should crash if the conbee is disconnected
i can just plug the conbee out, and the container doesnt stop/crash

so i think you are describing another issue, not related to this

**Comment by austwhite** [comment]
> for those on proxmox, mind looking on your syslog if the conbee 2 is getting reconnected? This issue happened to me [before](https://community.home-assistant.io/t/deconz-official-thread/100504/1890) on pi 4. Moved to proxmox and NUC and it got escalated to hassos crashing after updating it to 4.13 and upgrading the addon to 6.3.0. It turned out that it was an existing proxmox issue, see links on [870](https://github.com/home-assistant/operating-system/issues/870). It has been 10 days since I reinstalled the addon and replaced my conbee 2 stick, no crash yet.

I use to have this and fixed it by getting a powered USB Hub for the Conbee II.   I found connecting it directly to the PC would continuously disconnect.  I have not had a single disconnect of the Conbee stick since connecting to a powered hub.

**Comment by pergolafabio** [comment]
this issue is not about disconnecting, please create another thread

**Comment by austwhite** [comment]
> this issue is not about disconnecting, please create another thread

I can confirm this as well.  I have had many crashes, but no disconnects of the Conbee stick

**Comment by vukisz** [comment]
Hi. Are there any news on this? Just got another stop. And heating season is starting. My heating system sensors and switches is zigbee controlled. 
Or just ignore and Toggle watchdog on the deconz add-on?

**Comment by EmbeddedDevver** [comment]
Just for you to double check, I noticed with the recent OS update from HASSIO, that the COM port of the Conbee 2 stick was duplicated.  

  - /dev/ttyACM0  **(this was normally the good one)**
  - /dev/ttyACM1 **(Only this now is working well**
  - >-
    /dev/serial/by-id/usb-dresden_elektronik_ingenieurtechnik_GmbH_ConBee_II_DE1964682-if00 **(this also was normally working well, but now not anymore)**

While starting the plugin of deconz, I also noticed that there were mulitple Conbee sticks detected despite I only have 1 stick.

**Comment by austwhite** [comment]
@EmbeddedDevver 
I have not had that issue on my system at all.  My Conbee II only shows up once as one device.
I am running HA as a VM on ProxMox.
I do have a Raspberry Pi in a different location and I haven't had that problem on that System either.
The crashing of the DeConz Plugin seems to have stopped for me as of 0.115.0 though and O. 0.116.x it seems stable too

**Comment by pergolafabio** [comment]
Pls stay on topic, it has nothing todo with the stick, it's the container that's stops random

**Comment by EmbeddedDevver** [comment]
> Pls stay on topic, it has nothing todo with the stick, it's the container that's stops random

Please stay open for all causes. You cannot be sure that the stick is 100% not the problem as long as the problem is not resolved. The stick is part of entire system.

Anyway, I have the same setup and no problem whatsoever. Now running stable ever since. Maybe something wrong with your setup after the upgrade? Try a re-install of the system?

**Comment by pergolafabio** [comment]
There are different issues, this issue thread is about the container that's stops, the stick is connected while it happend
It's a problem in the container, so reinstall doesn't work/help resolve the issue

Stick related issues should have a seperate thread

**Comment by austwhite** [comment]
> There are different issues, this issue thread is about the container that's stops, the stick is connected while it happend
> It's a problem in the container, so reinstall doesn't work/help resolve the issue
> 
> Stick related issues should have a seperate thread

The stick, or something to do with recognising the stick, could cause the add-on to crash, so it could very well be part of the problem.   
It is worth keeping an open mind.  
That said, if multiple copies of your stick are showing up, but the app/container/plugin is not crashing, that should be considered as a different issue.
🙂

**Comment by hpapagaj** [comment]
Same problem here. I am experiencing random disconnect and crash of the addon.

<img width="580" alt="Screenshot 2020-11-16 at 21 58 15" src="https://user-images.githubusercontent.com/72989831/99307514-e5a49880-2856-11eb-80fa-17e7cfa0b15e.png">

**Comment by pergolafabio** [comment]
Don't stale, still not resolved

**Comment by bturrie** [comment]
Started having random Deconz crashes a couple of days back on rpi 3b+. I'm running supervised Home Assistant with Deconz from the addon store. Here's what's in my log file

Logger: pydeconz.websocket
Source: /usr/local/lib/python3.8/site-packages/pydeconz/websocket.py:81
First occurred: 7:04:06 PM (2 occurrences)

When this happens Deconz seems to be running. When I open the webserver motion is not being detected. A restart of Deconz seems to fix the problem. The Deconz logs have many repeats of this:

19:16:24:339 Device TTL 2968 s flags: 0x7
19:16:46:281 [INFO] - No button handler for: SML001 endpoint: 0x02 cluster: 0x0400 command: 0x0A payload[0]: 000
19:16:46:283 ZCL attribute report 0x001788010203DC17 for cluster: 0x0400, ep: 0x02, frame control: 0x08, mfcode: 0x0000 
19:16:51:094 [INFO] - No button handler for: SML001 endpoint: 0x02 cluster: 0x0406 command: 0x0A payload[0]: 000
19:16:51:095 ZCL attribute report 0x0017880104B7A284 for cluster: 0x0406, ep: 0x02, frame control: 0x08, mfcode: 0x0000 
19:16:54:242 [INFO] - No button handler for: SML001 endpoint: 0x02 cluster: 0x0406 command: 0x0A payload[0]: 000
19:16:54:244 ZCL attribute report 0x0017880104B7A284 for cluster: 0x0406, ep: 0x02, frame control: 0x08, mfcode: 0x0000 
19:17:16:016 [INFO] - No button handler for: RWL020 endpoint: 0x02 cluster: 0x0001 command: 0x0A payload[0]: 021
19:17:16:018 ZCL attribute report 0x00178801021AD017 for cluster: 0x0001, ep: 0x02, frame control: 0x18, mfcode: 0x0000

**Comment by redwoodimage** [comment]
still broke

**Comment by bturrie** [comment]
I finally gave up. Most of my zigbee stuff were Hue dimmers etc. So I just moved them back to the hue hub and abandoned Deconz.

    On Wednesday, March 3, 2021, 08:30:39 AM CST, redwoodimage <notifications@github.com> wrote:  
 
 


still broke

**Comment by vukisz** [comment]
I also have increased crashing of Deconz add on
Ha version  2021.02.11
Had two crashes during this week. And was jo crashes for half a year or something.

**Comment by pergolafabio** [comment]
I have enabled watchdog for now, not z solution, but a workaround

**Comment by jdbrookes** [comment]
Anyone else notice their Deconz addon stopping in the last hour?

I hadn't experienced this for a few months, but at least I could restart the Addon successfully.

**Comment by austwhite** [comment]
I use the watchdog to automatically restart mine but I checked the logs and no restarts or crashes on my system.
I am using a virtual machine on ProxMox so not sure if that makes a difference.

**Comment by pergolafabio** [comment]
still happening without watchdog

**Comment by vukisz** [comment]
I still haven't encountered any issues after complete upgrade from RPI3 (HA on SSD) to RPI4 (HA on SSD). 
I instaliad Raspberry Pi OS on that old RPI3. It complained about low voltage. My power adapter was from a raspberry bundle from ALI express:
![image](https://user-images.githubusercontent.com/48272340/117946167-b8253d00-b317-11eb-9f2d-3527fdbcddc1.png)

For RPI4 I bought bundle from local electronics shop, and power adapter looks more genuine.
So be aware, that power adapter could be an issue in here.

**Comment by vassilis-panos** [comment]
The problem still exists on my two NUCs

**Comment by austwhite** [comment]
How are guys that have the issue running?   
Using HassOS on a Raspberry Pi or even on a VM is flawless now.  Not getting any stops.

Haven't tried other install methods.   I run HA more as an appliance situation now so not mucking around with other installs.

**Comment by pergolafabio** [comment]
When I had the issues i was running HassOs on a esxi...
But enabled watchdog

**Comment by outofsight** [comment]
since yesterday deconz addon stopped working

keeps crashing with messages

```
11:53:55:765 [INFO] - No button map for: SP 120, unicast to: 0x0000, endpoint: 0x01, cluster: 0x0B04, command: 0x0A, payload: 0B0529B400, zclSeq: 124
11:53:55:765 ZCL attribute report 0x00158D000389312A for cluster: 0x0B04, ep: 0x01, frame control: 0x18, mfcode: 0x0000 
[11:53:57] WARNING: Your direct VNC access is not protected!
[11:53:57] INFO: Starting VNC server (local/no)...
2021/08/08 11:53:57 [notice] 311#311: signal 15 (SIGTERM) received from 304, exiting
In exit
Closing socket listening at 127.0.0.1:5901
2021/08/08 11:53:57 [notice] 1368#1368: exiting
2021/08/08 11:53:57 [notice] 1368#1368: exit
2021/08/08 11:53:57 [notice] 311#311: signal 17 (SIGCHLD) received from 1368
2021/08/08 11:53:57 [notice] 311#311: worker process 1368 exited with code 0
2021/08/08 11:53:57 [notice] 311#311: exit
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.

```

whatchdog doens't help

**Comment by reprz** [comment]
I just witnessed a random deconz shutdown on my Raspberry 3 system. This had happened couple of times during last year so it's not that common.

**Comment by jdbrookes** [comment]
> I just witnessed a random deconz shutdown on my Raspberry 3 system. This had happened couple of times during last year so it's not that common.

Yes me too, about an hour ago on my Tinkerboard. This hasn't happened in several months, and it took 2 attempts to restart the Addon. Its very unusual, especially as the occurrence of the issue seems to be synchronised across the world!

**Comment by pfrenssen** [comment]
I also had this shutdown around 2 hours ago.

**Comment by SiliconAvatar** [comment]
As the original issue author I haven't commented recently since I assumed the problem was resolved since I have not had any recent problems, but mine just restarted today as well (4:08pm EST) just the same time as the first new post here.

However mine will not start again, experiencing the same issue as already reported (below), where it won't stay running now.
https://github.com/home-assistant/addons/issues/2155

Since at least 4 people have had the container crash at approximately the same time, again, this is clearly an issue with the addon, and not just something with our specific system.

**Comment by xADDRx** [comment]
I have the same problem. Addon stop itself. Last day I try to start it but it doesn't start. Now it works. I don't know where ist the problem....

**Comment by reprz** [comment]
I had to start Deconz addon two times before it stayed on. After seeing other responds looks like this is not a local issues at all.

**Comment by maxoss** [comment]
same issue here. Random stops deconz addon

**Comment by pergolafabio** [comment]
> I use deconz and have not had it stip responding for months. What situation is it stopping in now? What version of Home Assistant and what configuration? I am using Home Assistant OS in a virtual machine and deconz is rock solid now on latest versions

I run hassos too, but you probably don't notice  if you have enabled the watchdog...

**Comment by pfrenssen** [comment]
Not stale.

**Comment by JBusch** [comment]
Having the same problem. Also, supervisor is crashing. If that happens, everything else works but deconz and supervisor page is unavailable

**Comment by JBusch** [comment]
deconz is not usable for the whole month now. anyone addressing that issue? happily provide logs or help to debug.

**Comment by austwhite** [comment]
@JBusch what set up do you have?  I have had no issues with Deconz for months.
 I use Deconz with a conbee2 and have absolutely no problems using the latest HA release and the latest deconz add on.
I am using HA OS (HomeAssistant) in an Open box VM with USB directly passed through.
I have a second install.with a ConBee2 on a Home Assistant Blue that also works flawlessly.
What specific setups are not working?
Aside from my install, I have 3 other installs also running Deconz no issue.   All are running HA OS

**Comment by JBusch** [comment]
@austwhite I'm running supervised HA OS on a RPI 4 8GB with SSD. When deconz crashes, supervisor also crashes. so you think i should open another issue? deconz is crashing randomly which is pretty much the description of this topic?

**Comment by austwhite** [comment]
The original issue this was for never crashed supervisor.  Crashing supervisor is definitely different.  I had the Deconz issue when this was first raised also.  Never once had supervisor crash

**Comment by JBusch** [comment]
Ok, i'll raise a new issue. Thanks!

**Comment by pergolafabio** [comment]
Unstale

**Comment by outofsight** [comment]
Again. Very frustrating. From time to time I'm affected by this issue. It was working for months and happened again march 9 2022, later worked for a week and today march 16 again. Every time in the middle of the night. Zigbee Smart plugs and heating become crazy.

I noticed that often this misbehavior is related with a pending HA Core update and often installing the update solve. It is possible?

Raspberry Pi 4 booting from SSD USB3 with Debian 11 and HA Supervised

**Comment by denisj07** [comment]
Same here, Deconz container stopped working this night and a HA Core update was available.

**Comment by elchininet** [comment]
> Same here, Deconz container stopped working this night and a HA Core update was available.

Same here, this night at 2:29 AM CET I faced this issue for the first time. HA OS on Raspberry Pi4 8GB with an SSD. I just started the addon and I needed to restart HA.

In my case two updates are available: HA Core and HA OS

**Comment by outofsight** [comment]
For me, restarting (addon, core, supervisor, host) is not enough. Deconz keeps crashing. Usually stop crashing after update. But the update is often breaking and I end up with unusable home automation until i fix the configuration.

**Comment by elchininet** [comment]
Yesterday, I updated core at night (from 2022.3.6 to 2022.3.8). Today at 12:05, Deconz have shut down itself, I noticed that there is another update available at the moment (HA OS 7.6).

**Comment by SiliconAvatar** [comment]
Mine crashed this morning as well (5:26am EST), first time it's happened in a while.

**Comment by austwhite** [comment]
This issue is closed.  If the error is regular you may need to open another one.
Can I ask if though what setup the issue is happening on?
Since I moved to Home Assistant Blue and stopped using the VM I have had absolutely no issues.
I found in a VM or Docker container, often the hardware would drop out.  The Conbee II would actually disappear.  No other USB devices disappear, including the xbee stick I use in a test ZHA mesh.
I think this needs reporting to Deconz as a possible hardware or firmware fault and that the add-on actually doesn't have any issues

**Comment by SiliconAvatar** [comment]
@austwhite ,

I realize this is a long thread, with 150 replies, but if you look back you'll see this happens with all different hardware and setups. It's not easily isolated to a single item such as only VMs or only RasPis, etc. That's why (I presume) no maintainer has ever tried to resolve this in the almost 2 years it's been an issue. Their only solution was to enable the Watchdog setting in the addon once that option was created.

It's not a hardware issue, considering it occurs on a vast array of different hardware. 
The problem you had of the conbee being disconnected is an entirely separate issue from this. For this issue, restarting the addon manually (or automatically via the Watchdog) fixes it *most* of the time. The conbee never becomes disconnected.

**Comment by pergolafabio** [comment]
Indeed , it's not a hardware issue..

I even remember that the addon stopped at the same exact time for several users...
Since that time, indeed the watchdog was introduced... So I don't see the issue anymore... But it's probably still happening

**Comment by austwhite** [comment]
It's been a long time since the watchdog was introduced, over a year if I recall, and the add on has not crashed for as many users since then.
As this issue has been closed, it would be best top open a new one.
I have only had it crash recently when the hardware disappears.  
Have you tried running Deconz and Phoscon seperate from Home Assistants addon too see if the same issue occurs?

**Comment by outofsight** [comment]
In my situation watchdoc has always been useless. Whet deconz starts to crash, keeps crashing and being restarted by watchdog and again crashing. Even for days...

**Comment by austwhite** [comment]
If none was still that unstable, I would consider going over to zha.   I know when I was using the VM I almost did that due to the Conbee always disappearing

**Comment by outofsight** [comment]
Again today for me with 2022.4 update pending.
I opened another issue here https://github.com/home-assistant/addons/issues/2430
